### PR TITLE
GH-40734: [Packaging][Debian] Drop support for Debian bullseye

### PR DESCRIPTION
**Pull request by raulcd**
### Rationale for this change

Debian GNU/Linux bullseye will reach EOL on 2024-07: https://wiki.debian.org/DebianReleases

We can drop support for it after we release 16.0.0 because 17.0.0 will be released after 2024-07.

### What changes are included in this PR?

Remove Debian bullseye support

### Are these changes tested?

Yes on CI.

### Are there any user-facing changes?

Yes, no support for Debian bullseye but no breaking changes on code.
* GitHub Issue: #40734

### CODE REVIEW

**Comment by raulcd** [comment]
@github-actions crossbow submit -g packaging

**Review by kou — APPROVED** [review]
+1

File: r/tools/nixlibs.R [hunk 1]
```diff
@@ -386,9 +386,7 @@ distro <- function() {
   out$id <- tolower(out$id)
   # debian unstable & testing lsb_release `version` don't include numbers but we can map from pretty name
   if (is.null(out$version) || out$version %in% c("testing", "unstable")) {
-    if (grepl("bullseye", out$codename)) {
-      out$short_version <- "11"
-    } else if (grepl("bookworm", out$codename)) {
+    if (grepl("bookworm", out$codename)) {
```
[the remark this answers is not in the document]
Comment (raulcd) [reply r/tools/nixlibs.R:389]: @jonkeane @assignUser thoughts on this? Can I merge this PR?

[the remark this answers is not in the document]
Comment (jonkeane) [reply r/tools/nixlibs.R:389]: Yeah, that's good to do here. It's not a huge concern if we had the `bullseye` one around, but ok to remove it. And good to have bookworm added.

Thanks!

**Pull request merged**